src/AWS.Deploy.CLI/Commands/GenerateDeploymentProjectCommand.cs

```diff
@@ -0,0 +1,237 @@
+            if (!string.IsNullOrEmpty(errorMessage))
+            {
+                if (newDirectoryCreated)
+                    Directory.Delete(saveCdkDirectoryPath);
+                throw new InvalidSaveDirectoryForCdkProject(errorMessage);
+            }
+
+            var directoryUnderSourceControl = await IsDirectoryUnderSourceControl(saveCdkDirectoryPath);
+            if (!directoryUnderSourceControl)
+            {
+                _toolInteractiveService.WriteLine();
+                var yesNoResult = _consoleUtilities.AskYesNoQuestion("The target directory for saving the CDK " +
+                    "deployment project is not being tracked by a source control system. Do you still want to continue?", YesNo.Yes);
+
+                if (yesNoResult == YesNo.No)
+                {
+                    if (newDirectoryCreated)
+                        Directory.Delete(saveCdkDirectoryPath);
+                    return;
+                }
+            }
+
+            await _cdkProjectHandler.CreateCdkProjectForDeployment(selectedRecommendation, _session, saveCdkDirectoryPath);
+            var directoryInfo = GetDirectoryInfo(saveCdkDirectoryPath);
+
+            _toolInteractiveService.WriteLine();
+            _toolInteractiveService.WriteLine($"The CDK deployment project is saved at: {directoryInfo.FullName}");
+        }
+
+        /// <summary>
+        /// Returns and instance of the Orchestartor class.
+        /// </summary>
+        /// <returns><see cref="Orchestrator"/></returns>
+        private Orchestrator GetOrchestrator()
+        {
+            return new Orchestrator(_session, new[] { RecipeLocator.FindRecipeDefinitionsPath() });
+        }
+
+        /// <summary>
+        /// This method generates the appropriate recommendations for the target deployment project.
+        /// </summary>
+        /// <param name="orchestrator"><see cref="Orchestrator"/></param>
+        /// <returns>A list of <see cref="Recommendation"/></returns>
+        private async Task<List<Recommendation>> GenerateDeploymentRecommendations(Orchestrator orchestrator)
+        {
+            var recommendations = await orchestrator.GenerateDeploymentRecommendations(forDeployment: false);
+            if (recommendations.Count == 0)
+            {
+                throw new FailedToGenerateAnyRecommendations("The project you are trying to deploy is currently not supported.");
+            }
+            return recommendations;
+        }
+
+        /// <summary>
+        /// This method takes the path to the target deployment project and creates
+        /// a default save directory inside the parent folder of the current directory.
+        /// For example:
+        /// Target project directory - C:\Codebase\MyWebApp\xyz.csproj
+        /// Generated default save directory - C:\Codebase\MyWebAppDeploymentProject If the save directory already exists, then a suffix number is attached.
```

Comment:
Lets use the default name format of `<app-project>.Deployment.csproj`. Periods help the name look more readable and a Project suffix in the name is redundant.

Reply:
Fixed